storage/innobase/trx/trx0trx.cc

```diff
@@ -725,9 +747,48 @@ dberr_t trx_lists_init_at_db_start()
 		return err;
 	}
 
+  if (srv_operation == SRV_OPERATION_NORMAL) {
+    if (trx_sys.is_start_from_scn() && !srv_use_scn) {
+      if (!trx_sys.is_undo_empty() || !purge_sys.purge_queue.empty()) {
```

Comment:
Is the check `purge_sys.purge_queue.empty()` really needed?

Reply:
Well, the check `purge_sys.purge_queue.empty()` is not needed here.